packages/rocketchat-cas/cas_client.js

```diff
@@ -33,7 +33,12 @@ Meteor.loginWithCas = function(options, callback) {
 	}
 
 	const appUrl = Meteor.absoluteUrl().replace(/\/$/, '') + __meteor_runtime_config__.ROOT_URL_PATH_PREFIX;
-	const loginUrl = `${ login_url }?service=${ appUrl }/_cas/${ credentialToken }`;
+	// check if the provided CAS URL already has some parameters
+	if (login_url.split('?').length > 1) {
+		const loginUrl = `${ login_url }&service=${ appUrl }/_cas/${ credentialToken }`;
```

Comment:
There is a issue here, since const is block scoped it will not exist outside the if/else blocks, so I recommend using a 
> let loginurl = [else value here]
and then changing it in case the if condition is true.